src/frequenz/sdk/microgrid/component_graph.py

```diff
@@ -748,6 +774,82 @@ def dfs(
 
         return component
 
+    def find_first_grid_source_component(
+        self,
+        categories: Iterable[ComponentCategory] | None = None,
+    ) -> Component:
+        """Find the first grid source component given a list of categories.
+
+        The order of the categories in the list determines the priority of the
+        component, with the first category having the highest priority.
+
+        If no categories are provided, the following categories are used in
+        priority order: `METER`, `INVERTER`, `EV_CHARGER`.
+
+        Args:
+            categories: The list of categories to search for the first grid
+                source component in.
+
+        Raises:
+            ValueError: when the component graph doesn't have a `GRID` component
+                or when no component is found in the given categories.
+
+        Returns:
+            The first grid source component found in the graph.
+        """
```

Comment:
This looks too specific for the component graph IMHO.

Can't we make this more generic, for example passing the type of component to find instead of hardcoding it to `Grid` and call it something like `find_first_descendant(self, root_category, descendant_categories)` instead? Or is this too specific to the grid to make any sense?

Reply:
At the moment the grid is the only root category. The suggestion still makes sense to me for future scenarios. I'll address the suggestion to make it more generic but probably set GRID category as a default value

Reply:
Well, the root  category doesn't need to be the root of the whole graph, maybe this is useful to find batteries attached to one inverter for example.

Reply:
True, then I won't set any default value